### ISSUE TITLE
DELETE key should perform 'Delete from Disk'

### ISSUE THREAD

**Issue by G-9999** [clementine-player/Clementine#6164]
Right now:
- Backspace key performs 'Remove from playlist'
- Delete key performs 'Remove from playlist'

Suggested behavior:
- Backspace key performs 'Remove from playlist' (no change)
**- Delete key should perform 'Delete from Disk'** (changed!)

Additionally, when the 'Delete from disk' action is started, the prompt that appears has two buttons: **Cancel** and **Yes**. I believe **Cancel** should be changed to **No** and a temporary key binding during this prompt should allow a **Y** or **N** key to be pressed to perform either of these actions. This would make deleting files from disk entirely possible with keyboard actions alone -- not requiring a mouse.

**Comment by amhndu** [comment]
Agreed. Alternative that doesn't break established behavior could be to at-least allow setting custom shortcuts for file-operations which includes Delete From Disk.
I can try to make a patch for this.

**Comment by G-9999** [comment]
@jonaski Don't forget that there is a confirmation dialogue before a file is deleted. Hitting DELETE by mistake would not do anything. This change would considerably speed the process of those who use Clementine to manage files/folders based music collections like myself! 🎃

**Comment by tomachinz** [comment]
This is a great idea! Library maintenance is why I'm trying to ditch iTunes and discovered Clementine!

Suggest **Yes** changed to **Delete**, with a D key instead. Personally I like cancel because it requires less thought to process what it might mean/do: it cancels which is never a bad thing unlike deleting a file. I'd leave that with cancel, and label the dialog if its unclear. Putting the file name into the delete dialog would be lovely if there is time, but not at the expense of getting this important improvement through.

> This would make deleting files from disk entirely possible with keyboard actions alone -- not requiring a mouse.

This is the most important thing ever, **keyboard use without the mouse** is number 1 most critical usability issue with any app; for sure it's why I'm looking into becoming a contributor on the project.

**Comment by jonaski** [comment]
@G-9999 I assume you have never had a cat.

I see no problems creating more shortcuts. But please leave the default for 'delete' to delete from playlist so we don't change the default behavior for what people are used to now.

**Comment by tomachinz** [comment]
> @G-9999 I assume you have never had a cat.
> 
> I see no problems creating more shortcuts. But please leave the default for 'delete' to delete from playlist so we don't change the default behavior for what people are used to now.

How is someone then to delete a file without a mouse?
A user on macOS has no solution except to press and hold the downward cursor 15 times. ITS REALLY HARD TO COUNT TO FIFTEEN personally, imagine someone in a wheelchair pushing a stick with their tongue just to do this....

ALSO pressing "D" does nothing with this right click menu open by the way. What os are you on? 

On mac I see this upon right click: 

![image](https://user-images.githubusercontent.com/1026291/46582327-4d496080-caa1-11e8-8ca8-c0c4fe3213af.png)

**Comment by tomachinz** [comment]
> @G-9999 I assume you have never had a cat.
> 
> I see no problems creating more shortcuts. But please leave the default for 'delete' to delete from playlist so we don't change the default behavior for what people are used to now.

Naturally, good UI would have cancel be the default for a delete operation. I also have a cat, so yeah delete then enter should cancel. Delete then D or Y would be marvellous. I got lots of files to delete. Deleting is why I installed Clementine! Love delete.

**Comment by G-9999** [comment]
@tomachinz that's what I see on the latest nightly builds on macOS too. However, even if you arrow key down to delete (15 times) the only keyboard control that seems to be accepted on the popup dialog is the ESC and ENTER keys, both of which Cancel. I think D is a better option than my original suggestion for that screen as well, since it is far away from DELETE and perhaps a cat won't hit one after another!

Just thinking more about this... would it make sense to add 3 more configurable shortcuts:

**Delete from Disk** (defaults to unset but user can set as DELETE or other)
**Delete from Disk Cancel** (defaults to ENTER)
**Delete from Disk Confirm** (defaults to unset, but user can set it to D or other)

_...OR..._

Just add one new shortcut:
**Delete from Disk** (defaults to unset but user can set as DELETE or other)
...and hard-code D and ENTER as the confirm/cancel keys on the dialog

Not sure which is better...?

**Comment by tomachinz** [comment]
Ah yes that is better. You could go one better make it like backspace upon delete, such that said track is removed from playlist for the keyboard challenged / caught out by the change. They'll still see something disappear but its from the playlist. I must admit I'm not a heavy user so I trust your OP motivation. I've not yet been able to get the app stable enough to consider de-duping.

**Comment by wheresvic** [comment]
Hi,

This is a little bit tangential to the current issue at hand but you can currently delete the currently playing song by sending a flag to clementine.

```
  -x, --delete-current      Delete the currently playing song
```

I programmed a global shortcut (i'm using ubuntu) to send this flag and it deletes the currently playing song. 

I would also be happy to have a delete from disk keyboard shortcut - I remember making a pull request even but it was rejected at the time.

**Comment by tomachinz** [comment]
Can you show me your pull request per chance?

Also, call me a noob, but how does one "send a flag to clementine"? (assume it's the internal event API the GUI plugs into).

I had a quick fish about the github repo but had trouble finding a list of these flags. Thanks in advance!

**Comment by wheresvic** [comment]
@tomachinz 

It is a program flag, if you simply do `clementine --help` you will see all the flags that you can pass to th program, e.g. `C:\Program Files (x86)\Clementine>clementine.exe --stop` and `C:\Program Files (x86)\Clementine>clementine.exe --play`

On ubuntu gnome you can setup some custom shortcuts via `Settings -> Devices -> Keyboard`. I setup a shortcut that will call `clementine -x` which then deletes the current song!

Regarding the PR, I cannot find it but I can make one again :)

**Comment by wheresvic** [comment]
@tomachinz https://github.com/clementine-player/Clementine/pull/6177
